packages/vuetify/src/components/VAutocomplete/VAutocomplete.tsx

```diff
@@ -132,7 +132,15 @@ export const VAutocomplete = genericComponent<new <
     const { filteredItems, getMatches } = useFilter(props, items, computed(() => isPristine.value ? undefined : search.value))
     const selections = computed(() => {
       return model.value.map(v => {
-        return items.value.find(item => props.valueComparator(item.value, v.value)) || v
+        return items.value.find(item => {
+          if (props.itemValue !== 'value' && props.returnObject) {
```

Comment:
It's only affected by return-object
```suggestion
          if (props.returnObject) {
```

Reply:
Something funny happens when we use `return-object` and the object doesnt have `value`

![image](https://github.com/vuetifyjs/vuetify/assets/18427030/812eaa4d-d742-4e97-8191-3a0b3e829bb7)

Code taken from first unit test in `VSelect`

```vue
<script setup>
import { ref } from 'vue'

const items = [
  { title: 'a' },
  { title: 'b' },
  { title: 'c' },
]
let model = ref([{ title: 'b' }])

</script>

<template>
  <v-app>
    <v-main>
      <VSelect
        :multiple="true"
        :return-object="true"
        :items="items"
        v-model="model"
        >
        <template #selection="{item, index}">
          {{item.raw.title.toUpperCase()}}
        </template>

      </VSelect>
    </v-main>
  </v-app>
</template>
```

Reply:
Saw that when using `return-object` and the object doesnt have `value` but only `title` it breaks.

I saw from Vuetify 2 the way the values are taken and compared to and used it here too

Would that make sense?

```diff
+ const getItemValue = (item: object) => getPropertyFromItem(item, props.itemValue, getItemTitle(item))
+ const getItemTitle = (item: object) => getPropertyFromItem(item, props.itemTitle, item.raw)
const selections = computed(() => {
  return model.value.map(v => {
    return items.value.find(item => {
      if (props.returnObject) {
        return props.valueComparator(
+        getItemValue(item.raw),
+         getItemValue(v.raw)
        )
      }
      return props.valueComparator(item.value, v.value)
    }) || v
  })
})
```